src/Scene.h

```diff
@@ -87,7 +105,8 @@ class CScene
 	Vec3f RayTrace(Ray& ray) const
 	{
 		// --- PUT YOUR CODE HERE ---
-		return Vec3f();
+		//return Vec3f();
+		return intersect(ray) ? ray.hit->getShader()->shade(ray) : m_bgColor;
 	}
```

Comment:
getShader() causes error here, because you never initialize ray.hit in the intersection algorithms